google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java

```diff
@@ -40,94 +34,51 @@
-    } else if (containingTable != null) {
-      // a table that is actually an element collection property
-      keyColumns = table::getColumnIterator;
-    } else {
-      // the case corresponding to a sequence-table that will only have 1 row.
-      keyColumns = Collections.emptyList();
-    }
-    return getTableString(table, metadata, keyColumns);
+  public String[] getSqlCreateStrings(Table currentTable, Metadata metadata) {
+    return buildSqlStrings(currentTable, metadata, true);
   }
 
-  private String[] getTableString(Table table, Metadata metadata, Iterable<Column> keyColumns) {
-
-    String primaryKeyColNames = StreamSupport.stream(keyColumns.spliterator(), false)
-        .map(Column::getQuotedName)
-        .collect(Collectors.joining(","));
-
-    StringJoiner colsAndTypes = new StringJoiner(",");
-
-    ((Iterator<Column>) table.getColumnIterator()).forEachRemaining(col -> colsAndTypes
-        .add(col.getQuotedName()
-            + " " + col.getSqlType(this.spannerDialect, metadata)
-            + (col.isNullable() ? this.spannerDialect.getNullColumnString() : " not null")));
-
-    ArrayList<String> statements = new ArrayList<>();
-    statements.add(MessageFormat.format(this.createTableTemplate, table.getQuotedName(),
-        colsAndTypes.toString(), primaryKeyColNames));
-
-    // Hibernate requires the special hibernate_sequence table to be populated with an initial val.
-    if (table.getName().equals(SequenceStyleGenerator.DEF_SEQUENCE_NAME)) {
-      statements.add("INSERT INTO " + SequenceStyleGenerator.DEF_SEQUENCE_NAME + " ("
-          + SequenceStyleGenerator.DEF_VALUE_COLUMN + ") VALUES(1)");
-    }
-
-    return statements.toArray(new String[0]);
+  @Override
+  public String[] getSqlDropStrings(Table currentTable, Metadata metadata) {
+    return buildSqlStrings(currentTable, metadata, false);
   }
 
-  private Table getContainingTableForCollection(Metadata metadata, Table collectionTable) {
-    for (Collection collection : metadata.getCollectionBindings()) {
-      if (collection.getCollectionTable().equals(collectionTable)) {
-        return collection.getTable();
-      }
-    }
-    return null;
+  /**
+   * Initializes the table exporter for if a new create-table or drop-table sequence is starting.
+   */
+  public void initializeTableExporter(Metadata metadata, boolean isCreateTables) {
+    tableDependencyTracker.initializeDependencies(metadata, isCreateTables);
   }
 
-  @Override
-  public String[] getSqlDropStrings(Table table, Metadata metadata) {
-    /* Cloud Spanner requires examining the metadata to find all indexes and interleaved tables.
-     * These must be dropped before the given table can be dropped.
-     * The current implementation does not support interleaved tables.
-     */
-
-    ArrayList<String> dropStrings = new ArrayList<>();
+  private String[] buildSqlStrings(Table currentTable, Metadata metadata, boolean isCreateTables) {
```

Comment:
Non-blocking personal opinion: an enum would be clearer than boolean for CREATE vs DROP.

Reply:
Agreed, but then you need to check for nulls and invalid options, etc.

Reply:
I decided to use the Action Enum provided in the Hibernate libraries (it has Action.CREATE, Action.DROP, etc.)

It'll be fine to use this; as far as invalid options/nulls, I think we will be able to avoid these problems since these functions are not intended to be called by end users. It is more to just make the code easier for us to understand with enums over booleans.